cms/templates/js/course-outline.underscore

```diff
@@ -57,11 +83,25 @@ if (staffOnlyMessage) {
         addStatusMessage(messageType, messageText);
     }
 
-    if (hasPartitionGroups) {
+    if (hasPartitionGroups && allowedGroups.length === 0 && xblockInfo.isVertical()) {
         addStatusMessage(
             'partition-groups',
             gettext('Access to some content in this unit is restricted to specific groups of learners')
         );
+    } else if (allowedGroups.length >= 1 && xblockInfo.isVertical()) {
+        var groupNames = [];
+        for (var i = 0; i < allowedGroups.length; i++) {
+            var groupName = groupIdToName[allowedGroups[i]];
+            groupNames.push(groupName);
+        }
+        messageType = 'partition-groups';
+        messageText = edx.StringUtils.interpolate(
+                gettext('Access to this unit is restricted to {groupNames}'),
+                {
+                    groupNames: groupNames.join(', ')
```

Comment:
This will not work for translation-- not all languages use commas in this way. You need to follow the same pattern as in contentstore/utils.py. Perhaps you can find a way to share the code.


````
# At the same time, build up all the selected groups as they are displayed in the dialog title.
    selected_groups_label = ''

    for index, partition in enumerate(selectable_partitions):
        for group in partition["groups"]:
            if group["selected"]:
                if len(selected_groups_label) == 0:
                    selected_groups_label = group['name']
                else:
                    # Translators: This is building up a list of groups. It is marked for translation because of the
                    # comma, which is used as a separator between each group.
                    selected_groups_label = _('{previous_groups}, {current_group}').format(
                        previous_groups=selected_groups_label,
                        current_group=group['name']
                    )